src/main/java/de/tum/in/www1/artemis/service/scheduled/cache/monitoring/ExamCache.java

```diff
@@ -1,12 +1,26 @@
 package de.tum.in.www1.artemis.service.scheduled.cache.monitoring;
 
+import java.util.function.UnaryOperator;
+
 import com.hazelcast.config.*;
 import com.hazelcast.core.HazelcastInstance;
 
 import de.tum.in.www1.artemis.config.Constants;
 import de.tum.in.www1.artemis.service.scheduled.cache.Cache;
 import de.tum.in.www1.artemis.service.scheduled.cache.CacheHandler;
 
+/**
+ * This class manages all {@link ExamMonitoringCache}s for all cached exams.
+ * <p>
+ * It encapsulates the Hazelcast objects to avoid unsafe actions on them and to prevent mistakes.
+ * Hazelcast distributed objects are more database like, which means that modifications of the objects themselves
+ * will not have any effect until they are send to all other instances, e.g. by replacing the value in the data structure.
+ * <p>
+ * To handle this better, we provide methods in this class that make {@linkplain #getReadCacheFor(Long) read-operations} and
+ * {@linkplain #getTransientWriteCacheFor(Long) write operations on transient properties} easier and less error prone;
+ * and that allow for {@linkplain #performCacheWrite(Long, UnaryOperator) atomic writes} (including an
+ * {@linkplain #performCacheWriteIfPresent(Long, UnaryOperator) if-present variant}).
```

Comment:
Where are these mentioned methods? Are they methods of Cache?

Reply:
In the CacheHandler :+1: but Stephan asked for a javadoc here as well (similar to the QuizCache)

Reply:
```suggestion
 * To handle this better, we provide methods in the {@linkplain #CacheHandler super class} that make {@linkplain #getReadCacheFor(Long) read-operations} and
 * {@linkplain #getTransientWriteCacheFor(Long) write operations on transient properties} easier and less error prone;
 * and that allow for {@linkplain #performCacheWrite(Long, UnaryOperator) atomic writes} (including an
 * {@linkplain #performCacheWriteIfPresent(Long, UnaryOperator) if-present variant}).
```
You might want to change this in the other sub class as well

Reply:
solved in 37deaeb6bfb31456f04f5ce03e851a015c2cfae8